dispatcher/subscription.go

```diff
@@ -77,26 +77,42 @@ func (sm *SubscriptionMap) MatchSubscribeEvent(event SubscribeEvent) {
 			DispatcherID: event.DispatcherID,
 			MatchLevel:   matchLevel,
 		}
-		sm.appendSubscription(subscription.Address, subscription)
+		sm.appendSubscription(subscription)
 	}
 }
 
-// Subscriptions returns the subscriptions map
-func (sm *SubscriptionMap) Subscriptions() map[string][]Subscription {
+// RemoveSubscriptions removes all subscriptions registered by a dispatcher
+func (sm *SubscriptionMapper) RemoveSubscriptions(dispatcherID uuid.UUID) {
+	sm.rwMut.Lock()
+	defer sm.rwMut.Unlock()
+
+	if _, ok := sm.subscriptions[dispatcherID]; ok {
+		delete(sm.subscriptions, dispatcherID)
+	}
+}
+
+// Subscriptions returns a slice reflecting the subscriptions present in the map
+func (sm *SubscriptionMapper) Subscriptions() []Subscription {
 	sm.rwMut.RLock()
 	defer sm.rwMut.RUnlock()
-	return sm.subscriptions
+
+	var subscriptions []Subscription
+	for _, sub := range sm.subscriptions {
+		subscriptions = append(subscriptions, sub...)
+	}
+
+	return subscriptions
 }
 
-func (sm *SubscriptionMap) matchLevelFromInput(subEntry SubscriptionEntry) string {
+func (sm *SubscriptionMapper) matchLevelFromInput(subEntry SubscriptionEntry) string {
 	hasAddress := subEntry.Address != "" && strings.Contains(subEntry.Address, erdTag)
 	hasIdentifier := subEntry.Identifier != ""
 	hasTopics := len(subEntry.Topics) > 0
 
 	if hasAddress && hasIdentifier && hasTopics {
 		return MatchTopics
 	}
-	if hasAddress && hasIdentifier {
+	if hasAddress && hasIdentifier || hasIdentifier{
```

Comment:
I'm not sure you need hasAddress. If hasIdentifier is false then if expression will be false. Also if hasAddress is false and hasIdentifier is true then if expression is true. This means that only hasIdentifier will have an impact on the if resolve

Reply:
done. added new matchLevel